### PR TITLE
Add card logos

### PR DESCRIPTION
**Pull request by huwshimi**

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- if you have a payment method open your account page
- scroll down to your card
- flip the card
- you should see a brand logo

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7114/

File: jujugui/static/gui/src/app/assets/svgs/card-american-express.svg [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
...KsYziSOiVeAz3MvVPvoX5RCN+HftpfM9H2hNPZ6B1nwRRJQb+4/snRj2hXAiBr8WOgAghI3EwJzkiBYzDXQlDkrfubxvroEI2XKzTPKj26HPRoXp9zujw4L+A+tPB/u/zAAzZ7NbQyHO+98G7mB3kT+gqgHHvAF/qf8lsi0bgK/6OaKwEwgL+1B7X3U+KjAn3BznbuCj5g9gLxdOq4F+jRVYCoCF1ezAsQXEDtDvThgiMtQ3sZ+jB4EZJGRIJeCvoFfchcDLyY6s3tCnaxdzgrAWClQOFvEX2tuduUYARP7RXvBF10L6apx9+HWayx6SB20KP3o3APU2c4K7jvTiXwEPrQfQqCZyHBTrm20IfMHwj9iPaQAGWEJR/Qu00Jal0kRjH7Gc0t0Omoj2IbgCMeAjdBD8miSOPXmLsAmZCtIEu+SeAC+tCdvTJzi7gIrOj/a6Afg1oJ4At59j3uFkzIClwXEMz+ww3ZcmjvGk8ZpIFteDvbwfPxUuZN0MPINe4hcIEwIZvB4P5zQj+GmSFytn2ZOKU4CDycB41xS1xs6Vv5d5grAbT8/9HAcZYPdHwTdswTqvis3Yn+potrUw8RD/WCuWKwW6DAw/W+wD0sFOj3GCQw6/rhXNp6S9QPmtexFeWH0I/hHKIrBh40vyf0GcpKAM77b1+/KhMo0fOB/Ous6AP9EBB3s1vOGVLL0KDb0VYMnpmC2MOVvsaWgCuPPRjJbKEfMaSycPMwJPkicR1hEf7XQz8GuBKAoy1r0POBYmdZoSl7Nj2keJO4wCUtdKyTXwa4dXdh/A7CrlElTLvrHlVDdq2rTcDtPaavgyz619CH8B/r2OOjqwngfaD7woB5lhXURW7gusAQxT7kfoZyGt8Gv/V7He0d0MPsQ0h16PcIJWNF/YOhfydUn6Vr9PPV380pcrNOa7DO/D70SoACgosS3peNsSIxQFyD8jNGdRbPha8y2C55f0VfwD3c9ZgfQn8n0AN/wHkYres79MH3JzdO4B8d/RjUSgAfOegOkYuUoG5f1GLzRzX7WLaRfCDoYV94dV/bKZKkCqFDzU2ZgFu9RT8FfTm9eNUTMnVzuhWJhDQccb4/8fVb+6rr+wfxp8c/Z+BHzYCqz48K4E+/f87Aj5yBPxPg/28gQ3/U7gEPAQCA02jH05wGWgAAAABJRU5ErkJggg=="></image>
```
Comment (Lukewh) [inline jujugui/static/gui/src/app/assets/svgs/card-american-express.svg]: pls no base64 embedded images

File: jujugui/static/gui/src/app/assets/svgs/card-diners-club.svg [hunk 2]
```diff
@@ -0,0 +1,608 @@
...
+        <dc:title></dc:title>
+      </cc:Work>
+    </rdf:RDF>
+  </metadata>
+  <g
+     inkscape:label="Layer 1"
+     inkscape:groupmode="layer"
+     id="layer1"
+     transform="translate(-171.27991,-177.06816)">
+    <image
```
Comment (Lukewh) [inline jujugui/static/gui/src/app/assets/svgs/card-diners-club.svg]: again with embedded. No point using SVG if it's just embedding a bitmap

File: jujugui/static/gui/src/app/assets/svgs/card-discover.svg [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+     inkscape:groupmode="layer"
+     id="layer1"
+     transform="translate(-184.1122,-197.44107)">
+    <image
+       y="197.44107"
+       x="184.1122"
+       width="35.454166"
+       height="22.225"
+       preserveAspectRatio="none"
+       xlink:href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAIYAAABUCAYAAABHj0+1AAAAAXNSR0IArs4c6QAAHk1JREFUeAHt
```
Comment (Lukewh) [inline jujugui/static/gui/src/app/assets/svgs/card-discover.svg]: O_O

File: jujugui/static/gui/src/app/assets/svgs/card-jcb.svg [hunk 4]
```diff
@@ -0,0 +1,99 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (Lukewh) [inline jujugui/static/gui/src/app/assets/svgs/card-jcb.svg]: Yay :)!

File: jujugui/static/gui/src/app/components/account/payment-methods/card/card.js [hunk 5]
```diff
@@ -34,6 +34,21 @@ YUI.add('account-payment-method-card', function() {
     },
 
     /**
+      Generate the card logo.
+
+      @method _generateLogo
+      @param {Object} e The click event from the checkbox.
```
Comment (Lukewh) [inline jujugui/static/gui/src/app/components/account/payment-methods/card/card.js]: no 1 letter vars allowed apparently. `evt`

File: jujugui/static/gui/src/app/assets/svgs/card-american-express.svg (hunk 1 above)
Comment (huwshimi) [inline jujugui/static/gui/src/app/assets/svgs/card-american-express.svg]: So I guess the advantage of this is we can just use the SvgIcon component to handle these as sprites, but it is a bit hacky, any other suggestions about how we could do this?

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7125/

File: jujugui/static/gui/src/app/assets/svgs/card-american-express.svg (hunk 1 above)
Comment (Lukewh) [inline jujugui/static/gui/src/app/assets/svgs/card-american-express.svg]: Where did these images come from? Can we not just find some full svg versions?

Comment (huwshimi) [inline jujugui/static/gui/src/app/assets/svgs/card-american-express.svg]: Spencer sourced them for me, he said that he couldn't find svgs for all of the logos. So the problem we have is that our spriting system only handles svgs and we have svg and png logos, I guess I could have a map of brand names to files and load the images from the server and resize them...

Comment (Lukewh) [inline jujugui/static/gui/src/app/assets/svgs/card-american-express.svg]: I'll talk to Spencer today @huwshimi see if we can sort out some nice svgs. We should definitely use svg's and avoid some crazy resizing thang.

**Comment by hatched** [comment]
I've asked @Lukewh to look into getting image assets for this branch as they don't really need to scale and gif's or the like will hopefully be smaller than svgs in his case anyways.

**Comment by Lukewh** [comment]
@huwshimi I just had a quick scout around and found the following:
https://assets.ubuntu.com/v1/40501d02-American_Express_logo.svg
https://assets.ubuntu.com/v1/4eb3cbc8-diners-club.svg
https://assets.ubuntu.com/v1/7455b85d-discover-logo.svg

The discover one is slightly different (no black border) but could be updated if required.

**Comment by huwshimi** [comment]
Thanks @Lukewh I've updated the svgs, ready for another review.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7169/

**Comment by huwshimi** [comment]
Still need one review and QA on this branch, thanks!

**Comment by Lukewh** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**